daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -100,17 +100,6 @@ func rotate(name string, maxFiles int) error {
 
 // backup renames a file from fromPath to toPath
 func backup(fromPath, toPath string) error {
-	if _, err := os.Stat(fromPath); os.IsNotExist(err) {
-		return err
-	}
-
-	if _, err := os.Stat(toPath); !os.IsNotExist(err) {
-		err := os.Remove(toPath)
-		if err != nil {
-			return err
-		}
-	}
-
 	return os.Rename(fromPath, toPath)
```

Comment:
run os.Rename() and if err != nil, then check if `err.(*os.LinkError).Err == syscall.ENOENT` then return os.ErrNotExist.